main.go

```diff
@@ -190,10 +216,17 @@ func computeRoutePrefix(prefix string, externalURL *url.URL) string {
 	return prefix
 }
 
-func interruptHandler(l net.Listener, logger log.Logger) {
+func exitHandler(l net.Listener, quitCh <-chan struct{}, logger log.Logger) {
 	notifier := make(chan os.Signal, 1)
 	signal.Notify(notifier, os.Interrupt, syscall.SIGTERM)
-	<-notifier
-	level.Info(logger).Log("msg", "received SIGINT/SIGTERM; exiting gracefully...")
+
+	select {
+	case <-notifier:
+		level.Info(logger).Log("msg", "received SIGINT/SIGTERM; exiting gracefully...")
+		break
+	case <-quitCh:
+		level.Warn(logger).Log("msg", "Received termination request via web service, exiting gracefully...")
```

Comment:
Generally, we don't capitalize log messages (even if they are almost sentences), i.e. use "received" and not "Received".